### ISSUE TITLE
[bug]: Expires value on 200 OK one second less then expires value on REGISTER request; causes ever-diminishing cycle

### ISSUE THREAD

**Issue by gmzaza** [asterisk/asterisk#702]
### Severity

Minor

### Versions

18.20.1

### Components/Modules

res_pjsip

### Operating Environment

Ubuntu 20.04.6 LTS, x86_64, 5.4.0-105-generic

### Frequency of Occurrence

Constant

### Issue Description

When sending a REGISTER request to Asterisk with a specified "expires" value in the "Contact" header, Asterisk generates its 200 response with "expires" value reduced by 1 second. e.g. In the example below, the client uses 180 seconds, and Asterisk responds with 179 seconds.

This leads the client that generated the REGISTER to then re-register 1 second sooner then it would otherwise. The cycle repeats, over and over, with the expiration dropping by 1 second upon each and every re-registrtion, until the minimum allowable registration interval is hit (60 seconds). When the client re-registers on 59 seconds, Asterisk then rejects the registration, on account of the expiration value being too low.

Example REGISTER request:
![reg_1_request](https://github.com/asterisk/asterisk/assets/78900459/8e2033ca-94e2-4104-b3b9-3fd931e6bdbb)

Example 200 response:
![reg_1_response](https://github.com/asterisk/asterisk/assets/78900459/0a5c5f90-9e03-4126-86cf-a4f9be7bb3c3)

### Asterisk Issue Guidelines

- [X] Yes, I have read the Asterisk Issue Guidelines

**Comment by seanbright** [comment]
> The cycle repeats, over and over, with the expiration dropping by 1 second upon each and every re-registrtion, until the minimum allowable registration interval is hit

So I _think_ I understand why it is replying with a lower expiration than the one in the `REGISTER` request (latency between the UAC and registrar) but I don't understand why that would affect subsequent registrations. Are you saying that the UAC will use 179, then 178, then 177, etc. in subsequent `REGISTER` requests?

Edit: If so, then I would say the client is broken and not Asterisk. There's nothing I can see in the RFC 3261 that indicates the client should use the `expires` parameter value from the register response for subsequent `REGISTER` requests.

**Comment by seanbright** [comment]
I don't think this is a bug but feel free to comment and we can re-open if necessary.
